### PR TITLE
Update `pom.xml` config.

### PR DESCRIPTION
**Pull request by howardlunatech**
The older format doesn't work for newer maven.

### CODE REVIEW

**Comment by fommil** [comment]
@amanjpro what do you think of this suggestion? Do we perhaps need two installation sections divided by maven version?

File: build_tools/maven.md [hunk 1]
```diff
@@ -20,27 +20,49 @@ Configure your `~/.m2/settings.xml` file so that maven is aware of the plugin gr
...
+      <releases>
+        <enabled>true</enabled>
+      </releases>
+      <snapshots>
+        <enabled>false</enabled>
+      </snapshots>
+    </repository>
+    ...
+  </repositories>
+  ...
```
Comment (amanjpro) [inline build_tools/maven.md]: I think this is too much details, we can just state that the `snapshots` repository should be added to the `pom.xml` file so the plugin can fetch the ensime-server dependency.

File: build_tools/maven.md [hunk 2]
```diff
@@ -20,27 +20,49 @@ Configure your `~/.m2/settings.xml` file so that maven is aware of the plugin gr
...
+    <pluginManagement>
+      <plugins>
+        <plugin>
+          <groupId>org.ensime.maven.plugins</groupId>
+          <artifactId>ensime-maven</artifactId>
+          <version>1.0.3</version>
+        </plugin>
+        ...
+      </plugins>
+    </pluginManagement>
```
Comment (amanjpro) [inline build_tools/maven.md]: Again, I think we should only give the coordinates of the plugin and the user should be able to figure out where to put it.

**Comment by fommil** [comment]
needs to be raised again due to conflicts, if you're still wanting to add this.

**Pull request closed without merging**